src/main/java/org/example/EchoServerSocket.java

```diff
@@ -0,0 +1,19 @@
+package org.example;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+
+public class EchoServerSocket {
+
+    public ServerSocket serverSocketCreator(int portNumber) {
```

Comment:
It will be more corporate/java native approach to use `Factory` pattern when we need a class that instantiates instances of other classes. 
Something Like
```
SocketServerFactory
- create(portNumber)
- ```

Reply:
Changes from `EchoServerSocket` to `ServerSocketFactory` in latest commit.

Reply:
Great! Can't really find the usage of this class other than in tests. Is it still needed?

Reply:
Agreed, no longer needed. Removed from the repo.